pkg/apis/syn/v1alpha1/gitrepo_types.go

```diff
@@ -49,16 +49,19 @@ type GitRepoSpec struct {
 // controller creating the template instance.
 type GitRepoTemplate struct {
 	// APISecretRef reference to secret containing connection information
-	APISecretRef corev1.SecretReference `json:"apiSecretRef"`
+	APISecretRef corev1.SecretReference `json:"apiSecretRef,omitempty"`
```

Comment:
Why `omitempty` (optional)?

Reply:
If the GitRepo is not managed by the operator, e.g. RepoType=unamanged, you should not be required to add an API key.